dataland-frontend/src/components/general/AlignedActivitiesDataTable.vue

```diff
@@ -0,0 +1,322 @@
+      frozenColumnDefinitions: [] as Array<{ field: string; header: string; frozen?: boolean; group: string }>,
+      mainColumnGroups: [] as Array<{ key: string; label: string; colspan: number }>,
+      mainColumnDefinitions: [] as Array<MainColumnDefinition>,
+      frozenColumnData: [] as Array<{
+        activity: Activity | undefined;
+        naceCodes: string[] | undefined;
+      }>,
+      mainColumnData: [] as Array<ActivityFieldValueObject>,
+    };
+  },
+  mounted() {
+    const dialogRefToDisplay = this.dialogRef as DynamicDialogInstance;
+    const dialogRefData = dialogRefToDisplay.data as {
+      listOfRowContents: Array<object | string>;
+      kpiKeyOfTable: string;
+      columnHeaders: { [kpiKeyOfTable: string]: { [columnName: string]: string } };
+    };
+    this.kpiKeyOfTable = dialogRefData.kpiKeyOfTable;
+    this.columnHeaders = dialogRefData.columnHeaders;
+
+    if (typeof dialogRefData.listOfRowContents[0] === "string") {
+      this.listOfRowContents = dialogRefData.listOfRowContents.map((o) => ({ [this.kpiKeyOfTable]: o }));
+    } else {
+      this.listOfRowContents = dialogRefData.listOfRowContents as Array<EuTaxonomyAlignedActivity>;
+    }
+
+    this.frozenColumnDefinitions = [
+      { field: "activity", header: this.humanizeHeaderName("activity"), frozen: true, group: "_frozen" },
+      { field: "naceCodes", header: this.humanizeHeaderName("naceCodes"), frozen: true, group: "_frozen" },
+    ];
+
+    this.mainColumnDefinitions = [
+      { field: "revenue", header: this.humanizeHeaderName("revenue"), group: "_revenue", groupIndex: 0 },
+      { field: "revenuePercent", header: this.humanizeHeaderName("revenuePercent"), group: "_revenue", groupIndex: 1 },
+
+      ...this.makeGroupColumns("substantialContributionCriteria"),
+      ...this.makeGroupColumns("dnshCriteria"),
+
+      {
+        field: "minimumSafeguards",
+        header: this.humanizeHeaderName("minimumSafeguards"),
+        group: "_minimumSafeguards",
+        groupIndex: 0,
+      },
+    ];
+
+    this.frozenColumnData = this.listOfRowContents.map((activity) => ({
+      activity: activity.activityName,
+      naceCodes: activity.naceCodes,
+    }));
+
+    this.mainColumnData = this.listOfRowContents
+      .map((col) => [
+        ...createRevenueGroupData(col),
+        ...createSubstantialContributionCriteriaGroupData(col),
+        ...createDnshCriteriaGroupData(col),
+        ...createMinimumSafeguardsGroupData(col),
+      ])
+      .flat();
+
+    this.mainColumnGroups = [
+      { key: "_revenue", label: "", colspan: 2 },
+      {
+        key: "substantialContributionCriteria",
+        label: this.humanizeHeaderName("substantialContributionCriteria"),
+        colspan: 6,
+      },
+      { key: "dnshCriteria", label: this.humanizeHeaderName("dnshCriteria"), colspan: 6 },
+      { key: "_minimumSafeguards", label: "", colspan: 1 },
+    ];
```

Comment:
Is there a more elegant way of entering the colspan, e.g. via length of corresponding objects that have already been filled? Looks like a pitfall to me which could easily be overlooked in case that future dataModel changes necessitate adapting these `mainColumnGroups`

Reply:
yeah, I was aware of that possibility. I'll try to define it using the column/group definitions.. be right back.

Reply:
Done.
Calculating `substantialContributionCriteria` and `dnshCriteria` colspans via `EnvironmentalObjective` length.
`_revenue` and `_minimalSafeguard` groups don't have this info (not that I could find) and we create these groups/columns manually anyway, so I've left static values.